chia/full_node/full_node.py

```diff
@@ -633,6 +633,11 @@ async def _sync(self):
                 await weight_proof_peer.close(600)
                 raise RuntimeError(f"Weight proof had the wrong weight: {weight_proof_peer.peer_host}")
 
+            # dont sync to wp if local peak is heavier,
+            # dont ban peer, we asked for this peak
+            if response.wp.recent_chain_data[-1].reward_chain_block.weight <= self.blockchain.get_peak().weight:
+                raise RuntimeError(f"current peak is heavier than Weight proof peek: {weight_proof_peer.peer_host}")
```

Comment:
Does this throw only interrupt the WP validation process / thread, not the main networking thread?

Reply:
this is on the main thread  before we start the thread